File: 프로그래머스/lv0/120924. 다음에 올 숫자/다음에 올 숫자.java
class Solution {
    public int solution(int[] common) {
        int answer = 0;
        int lastElement = common[common.length - 1];

        if (isArithmeticSequence(common)) {
            int commonDifference = common[1] - common[0];
            answer = lastElement + commonDifference;
        } else {
            int commonRatio = common[1] / common[0];
            answer = lastElement * commonRatio;
        }

        return answer;
    }

    private boolean isArithmeticSequence(int[] sequence) {
        int commonDifference = sequence[1] - sequence[0];
        for (int i = 1; i < sequence.length; i++) {
            if (sequence[i] - sequence[i - 1] != commonDifference) {
                return false;
            }
        }
        return true;
    }
}

File: 프로그래머스/lv0/120924. 다음에 올 숫자/다음에 올 숫자.kt
class Solution {
    fun solution(common: IntArray): Int {
        var answer = 0
        val lastElement = common[common.size - 1]

        if (isArithmeticSequence(common)) {
            val commonDifference = common[1] - common[0]
            answer = lastElement + commonDifference
        } else {
            val commonRatio = common[1] / common[0]
            answer = lastElement * commonRatio
        }

        return answer
    }

    private fun isArithmeticSequence(sequence: IntArray): Boolean {
        val commonDifference = sequence[1] - sequence[0]
        for (i in 1 until sequence.size) {
            if (sequence[i] - sequence[i - 1] != commonDifference) {
                return false
            }
        }
        return true
    }
}
